SerialPrograms/Source/PokemonSV/Programs/Farming/PokemonSV_BlueberryCatchPhoto.cpp

```diff
@@ -0,0 +1,1028 @@
+                        std::string ball_reader = "";
+                        WallClock start = current_time();
+
+                        console.log("Opening ball menu...");
+                        while (ball_reader == "") {
+                            if (current_time() - start > std::chrono::minutes(2)) {
+                                console.log("Timed out trying to read ball after 2 minutes.", COLOR_RED);
+                                throw OperationFailedException(
+                                    ErrorReport::SEND_ERROR_REPORT, console,
+                                    "Timed out trying to read ball after 2 minutes.",
+                                    true
+                                );
+                            }
+
+                            //Mash B to exit anything else
+                            pbf_mash_button(context, BUTTON_B, 125);
+                            context.wait_for_all_requests();
+
+                            //Press X to open Ball menu
+                            pbf_press_button(context, BUTTON_X, 20, 100);
+                            context.wait_for_all_requests();
+
+                            VideoSnapshot screen = console.video().snapshot();
+                            ball_reader = reader.read_ball(screen);
+                        }
+
+                        console.log("Selecting Quick Ball.");
+                        int quantity = move_to_ball(reader, console, context, "quick-ball");
+                        if (quantity == 0) {
+                            //Stop so user can check they have quick balls.
+                            console.log("Unable to find Quick Ball on turn 1.");
+                            throw OperationFailedException(
+                                ErrorReport::SEND_ERROR_REPORT, console,
+                                "Unable to find Quick Ball on turn 1.",
+                                true
+                            );
+                        }
+                        if (quantity < 0) {
+                            console.log("Unable to read ball quantity.", COLOR_RED);
+                        }
+
+                        //Throw ball
+                        console.log("Throwing Quick Ball.");
+                        pbf_mash_button(context, BUTTON_A, 150);
+                        context.wait_for_all_requests();
+
+                        quickball_thrown = true;
+                        pbf_mash_button(context, BUTTON_B, 900);
+                        context.wait_for_all_requests();
+                    }
+                    else {
+                        BattleBallReader reader(console, BBQ_OPTIONS.LANGUAGE);
+                        std::string ball_reader = "";
+                        WallClock start = current_time();
+
+                        console.log("Opening ball menu...");
+                        while (ball_reader == "") {
```

Comment:
A lot of duplicated code here between the Quick Ball and normal ball paths. Can you pull this out into a separate function where you just pass in the ball to use?